lib/io/can.hpp

```diff
@@ -1,8 +1,33 @@
+#include <string>
+
+#include <core/logger.hpp>
 #include <core/types.hpp>
 
+#ifndef LINUX
+#include <linux/can.h>
+#define CanFrame can_frame
+#else
+struct CanFrame {
```

Comment:
You want these within the namespace so the types are actually accessibly outside this file